lib/date_time/date_time_core.c

```diff
@@ -106,11 +106,10 @@ static void date_time_core_schedule_retry(void)
 		return;
 	}
 
-	/* Scheduling new update cannot fail because we are never doing retries
-	 * if we have fresh enough time
-	 */
-	date_time_core_schedule_work(CONFIG_DATE_TIME_RETRY_INTERVAL_SECONDS);
-	LOG_DBG("Date time update retry in: %d seconds", CONFIG_DATE_TIME_RETRY_INTERVAL_SECONDS);
+	if (date_time_core_schedule_work(CONFIG_DATE_TIME_RETRY_INTERVAL_SECONDS) == 0) {
+		LOG_DBG("Date time update retry in: %d seconds",
+			CONFIG_DATE_TIME_RETRY_INTERVAL_SECONDS);
+	}
```

Comment:
Looks like you still need to either remove or squash in this revert commit

Reply:
Yes, I will remove or squash the revert commit depending on the outcome of this conversation:
https://github.com/nrfconnect/sdk-nrf/pull/17477#discussion_r1774942164

Reply:
Conclusion is remove the revert commit, I think. You were right all along!